Feat: add *whitespaces* characters, export symbol

kilianmh opened the pull request:
- added: `#\Vt` `#\Next-Line` `#\No-break_space`
- also: order characters depending on their char-code

Two further questions might be:
1. Are there any more whitespace characters?
2. The char-code of `#\Linefeed` `#\Newline` (10) is identical and `#\Linefeed` even evaluates to `#\Newline`. So can `#\Linefeed` be discarded from the list?

kilianmh committed:
Feat: add whitespace characters

- added: #\Vt #\Next-Line #\No-break_space
- also: order whitespace characters depending on their char-code

vindarel commented:
> The char-code of #\Linefeed #\Newline (10) is identical and #\Linefeed even evaluates to #\Newline. So can #\Linefeed be discarded from the list?

I would keep the two and add a comment. Probably on some platforms they are different.

LGTM.

kilianmh committed:
Docs: whitespaces char codes

Add docstring to *whitespaces* that linefeed and newline
represent the same character (code) on some implementations.

kilianmh commented:
Now there's some documentation added.

2 more things came things came to my mind regarding `whitespaces:`

- Should we export the symbol, because someone might want to use it outside of `cl-str`?
- Since `whitespaces` is basically a constant value, could/should it be defined as a constant (`alexandria:define-constant`) or global variable (`global-vars:define-global-var`) ? Then there would be no run-time boundness checks (for global-vars at least on SBCL, CCL, Lispworks) and as constant even value directly in place inserted at compilation (if i remember correctly).

vindarel commented:
> Now there's some documentation added.

\+1 

> 
> 2 more things came things came to my mind regarding `whitespaces:`
> 
>     * Should we export the symbol, because someone might want to use it outside of `cl-str`?

good idea

> 
>  Since `whitespaces` is basically a constant value, could/should it be defined as a constant 
> (`alexandria:define-constant`) or global variable (`global-vars:define-global-var`) ? 
> Then there would be no run-time boundness checks (for global-vars at least on SBCL, CCL, Lispworks) 
> and as constant even value directly in place inserted at compilation (if i remember correctly).

I was saying: I'd rather not include another dependency (and AFAIK Alexandria is not a transitive dependency. The only one is cl-ppcre (and change-case, but that's cl-ppcre)).

but `*whitespaces*` is not a constant value, a user could "let over" it (more so if it is exported).

kilianmh committed:
Docs: update whitespace chars in trim example

kilianmh committed:
Feat: export dynamic variable *whitespaces*

kilianmh commented:
Now `*whitespaces*` is an exported symbol. Also the whitespace characters in the `string-trim` examples in README are updated to the new character list.

> I'd rather not include another dependency

Yeah it's probably good to keep the transitive dependency list small.

> `*whitespaces*` is not a constant value, a user could "let over" it (more so if it is exported)

Makes sense

vindarel commented:
Very nice again, thanks.
